packages/mp/src/mp/core/unix.py

```diff
@@ -434,5 +434,70 @@ def check_lock_file(project_path: pathlib.Path) -> None:
         raise NonFatalCommandError(error_output) from e
 
 
+def get_changed_files_from_main(
+    base: str, head_sha: str, integration_path: pathlib.Path
+) -> list[pathlib.Path]:
+    """Return a list of file names changed in a pull request compared to the main branch.
+
+    Args:
+        base: The base branch of the PR.
+        head_sha: The head commit SHA of the PR.
+        integration_path: The path to the integration directory.
+
+    Returns:
+        A list of changed file paths.
+
+    Raises:
+        NonFatalCommandError: If the git command fails.
+
+    """
+    command: list[str] = [
+        "/usr/bin/git",
+        "diff",
+        f"origin/{base}...{head_sha}",
+        "--name-only",
+        "--diff-filter=ACMRTUXB",
+        str(integration_path) + "/",
+    ]
+    try:
+        results: sp.CompletedProcess[str] = sp.run(  # noqa: S603
+            command, check=True, text=True, capture_output=True
+        )
+        return [pathlib.Path(path) for path in results.stdout.split("\n") if path]
```

Comment:
Let's perform a check that the path indeed exists
```suggestion
        return [p for path in results.stdout.strip().splitlines() if path and (p := pathlib.Path(path)).exists()]
```